.github/workflows/pull-request.yml

```diff
@@ -12,16 +14,73 @@ jobs:
     name: Build Newlib
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
       with:
         submodules: true
-    - uses: actions/setup-python@v2
+
+    - uses: actions/setup-python@v4
       with:
         python-version: '3.x'
-    - name: Config and Compile
+
+    - name: Toolchain
+      run: |
+        mkdir -p $HOME/toolchain/
+        pushd $HOME/toolchain/
+        for fremote in "package/package_esp8266com_index.template.json" "tools/get.py" ; do
+          flocal="${fremote##*/}"
+          curl \
+            -H "Authentication: ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.VERSION.raw" \
+            -o "$flocal" \
+            https://api.github.com/repos/esp8266/Arduino/contents/$fremote
```

Comment:
I don't think we need any authentication here.  This is only ever getting the master version of the 2 files, no?